epidose/device/util.sh

```diff
@@ -129,6 +136,50 @@ run_python()
   "$PYTHON" "$SCRIPT_DIR"/"$prog".py $DEBUG_FLAG $VERBOSE_FLAG "$@"
 }
 
+# Check the validity of the cuckoo filter
+# preconditions: none 
+# Returns the time required to sleep (in seconds)
+# until the Cuckoo filter becomes stale.
+# If 0 is returned, the Cuckoo filte is stale.
+check_cuckoo_filter()
+{
+  if [ -r "$FILTER" ] ; then
+    filter_mtime=$(stat -c '%Y' "$FILTER")
+    time_now=$(date +%s)
+    filter_age=$((time_now - filter_mtime))
+    if [ $filter_age -lt $MAX_FILTER_AGE ] ; then
+      log "Filter's age ($filter_age s) is current; no update required"
+      to_sleep=$((MAX_FILTER_AGE - filter_age))
+    else
+      log "Filter's age ($filter_age s) makes it stale; update required"
+      to_sleep=0
+    fi
+  else
+    log "No filter available; download required"
+    mkdir -p "$(dirname $FILTER)"
+  fi	
+}
+
+# Obtain a (new) version of the Cuckoo filter
+# preconditions: none 
+# Returns an exit code that defines
```

Comment:
Similarly, an exit code has a specific meaning, it's not a global variable.  Use `return` and modify the function's name accordingly to reflect the returned value (`is_...`)